### PR TITLE
chore(E2E): add toBeExpanded() aria-expanded matcher

### LINKED ISSUE #2566: E2E: Set up custom matcher for `aria-expanded`
We have multiple instances where we're checking for the value of the [`"aria-expanded"` attribute](https://github.com/search?q=repo%3Aconfluentinc%2Fvscode+%22aria-expanded%22+path%3Atests%2F&type=code) (including in https://github.com/confluentinc/vscode/pull/2557), and that will continue to be common when we deal with views and (expandable) tree items. 

We should implement a [custom matcher](https://playwright.dev/docs/test-assertions#add-custom-matchers-using-expectextend) to clean those up. Ideally we could just call:
```ts
await expect(<locator>).toBeExpanded()
```

### PR DESCRIPTION
**Pull request by shouples**
## Summary of Changes

Across the E2E suite we repeatedly assert that a tree item or view is expanded by checking its `aria-expanded` attribute directly: `await expect(locator).toHaveAttribute("aria-expanded", "true")`. This PR adds a small Playwright custom matcher so that reads as `await expect(locator).toBeExpanded()` instead, and converts the 19 existing positive assertions to use it.

No functional or behavioral changes to any test. Full E2E pipeline [passing](https://semaphore.ci.confluent.io/workflows/97684cb7-eabd-4321-a434-fcf5591ad770?pipeline_id=7015f244-97b9-42ef-bd9a-bd7e89f70c8b). 

Closes #2566

### Optional: Any additional details or context that should be provided?

- The matcher lives in `tests/e2e/matchers.ts`, registers on Playwright's shared `expect` via `expect.extend`, and is wired in through a side-effect import in `tests/e2e/baseTest.ts` (which every spec already imports), so it's available everywhere without per-file setup. It's typed via a global augmentation so `.toBeExpanded()` type-checks anywhere `expect` is imported, and it delegates to `toHaveAttribute` internally, so auto-waiting and `.not.toBeExpanded()` still work.
- Two categories of `aria-expanded` usage were intentionally left as-is: the two `not.toHaveAttribute("aria-expanded")` checks (which assert the attribute is *absent* on a non-expandable leaf item, a stricter condition than "not expanded"), and the five `getAttribute("aria-expanded")` reads that feed an `if` to decide whether to click (control-flow, not assertions). A value-asserting matcher can't cover either without weakening or reshaping them.
- The matcher's own logic (both `toBeExpanded()` and `.not.toBeExpanded()`, across expanded / collapsed / absent states) is covered by a focused functional test at `src/webview/aria-expanded-matcher.spec.ts`. The rollwright functional harness is the only non-Electron way to run real-DOM Playwright assertions here, so the matcher's edge cases run fast in `npx gulp functional` without needing Docker/sidecar. The 19 converted assertions additionally exercise the positive path in the E2E suite in CI.
- Verified locally with `npx gulp check` (the type augmentation and all 19 converted call sites compile), `npx gulp lint`, `prettier`, and `npx gulp functional -t "toBeExpanded"` (3 passing).

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

#### Tests

- [x] Added new
- [x] Updated existing
- [ ] Deleted existing

#### Release notes

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md)?

### CODE REVIEW

File: tests/e2e/matchers.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  locator: Locator,
+  options?: ToBeExpandedOptions,
+): Promise<MatcherReturnType> {
+  const assertionName = "toBeExpanded";
+  let pass: boolean;
+  let matcherResult: { actual?: unknown } | undefined;
+  try {
+    // delegate through `.not` when negated so auto-waiting polls in the expected direction
+    const assertion = this.isNot ? expect(locator).not : expect(locator);
+    await assertion.toHaveAttribute("aria-expanded", "true", options);
```
[the remark this answers is not in the document]
Comment (shouples) [reply tests/e2e/matchers.ts:26]: Good catch. Added a focused functional test at `src/webview/aria-expanded-matcher.spec.ts` covering both `toBeExpanded()` and `.not.toBeExpanded()` across expanded, collapsed, and absent `aria-expanded` states. It runs on the rollwright harness (no Electron/Docker needed), and I confirmed it actually guards the inversion logic: removing the `pass = !pass` flip makes both `.not` cases fail. Done in 19fc013f.
